extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -684,7 +662,9 @@ export class AmpLightboxViewer extends AMP.BaseElement {
         this.element.ownerDocument.body.appendChild(transLayer);
         const rect = layoutRectFromDomRect(sourceImage
             ./*OK*/getBoundingClientRect());
-        const imageBox = this.getCurrentElement_().imageViewer.getImageBox();
+
+        const imageBox = /**@type {?}*/ (this.getCurrentElement_().imageViewer)
```

Comment:
Since the image viewer is a full custom element now, you should be able to access `getImageBox` directly if you have a reference to its DOM element.

Reply:
I'm confused about how this works. From `amp-carousel`'s usage, I thought for all full on components, you had to access its `implementation_` in order to get at its API. Is `amp-carousel` a special case here then? I'll play around with this.

Reply:
Yeah, if you remove the type cast and the `implementation`, it seems to error out with: 
```
error.js:190 TypeError: _this11.getCurrentElement_(...).imageViewer.getImageBox is not a function
```

Reply:
Hmm alright, what you have works then. I will need to learn more about this : /